packages/jellyfish-transaction-builder/__tests__/provider.mock.ts

```diff
@@ -43,14 +43,29 @@ export class MockPrevoutProvider implements PrevoutProvider {
     const address = Bech32.fromPubKey(pubKey, 'bcrt')
 
     // TODO(fuxingloh): minimumSumAmount behavior is weirdly inconsistent, listunspent will always
-    //  return the correct result without providing options. However, with 'minimumSumAmount', it
-    //  will appear sometimes. Likely due to race conditions in bitcoin code,
-    //  e.g. -reindex when importprivkey.
+    // return the correct result without providing options. However, with 'minimumSumAmount', it
+    // will appear sometimes.
+    // Refer -> https://github.com/DeFiCh/ain/issues/1208
+    // Due to the above, we collect all utxos for the given address here and extract what's required to statisfy the minBalance
+    // in jellyfish code.
     const unspent: any[] = await this.container.call('listunspent', [
       1, 9999999, [address], true
     ])
 
-    return unspent.map((utxo: any): Prevout => {
+    // extract utxos to satisfy minBalance
+    const utxoSelected: any[] = []
+    let total = new BigNumber(0)
+
+    for (const utxo of unspent) {
+      if (total.isGreaterThan(minBalance)) {
+        break
+      }
+
+      total = total.plus(utxo.amount)
+      utxoSelected.push(utxo)
+    }
```

Comment:
hmm... should fix https://github.com/DeFiCh/ain/issues/1208 first
then just 
```ts
const unspent: any[] = await this.container.call('listunspent', [
  1, 9999999, [address], true, { minimumSumAmount: minBalance }
])
```
? :thinking:

Reply:
Maybe this is a temporary workaround while waiting for 1208 to be fixed later.

Reply:
yeah, #1208 requires changing of internal functions and cpp team's attention. This is a workaround until then.